### ISSUE TITLE
Sonoma

### ISSUE THREAD

**Issue by ofirgalcon** [munkireport/mdm_status#19]
I see this error

```File "/usr/local/munkireport/scripts/mdm_status.py", line 191, in
main()
File "/usr/local/munkireport/scripts/mdm_status.py", line 177, in main
result['last_mdm_kickstart'] = string_to_timestamp(watchdog_state[item])
File "/usr/local/munkireport/scripts/mdm_status.py", line 137, in string_to_timestamp
dt_utc = datetime.strptime(date_str.strip(), "%Y-%m-%dT%H:%M:%S.%f") # 2023-08-17T01:21:54.555443-04:00
File "/Library/ManagedFrameworks/Python/Python3.framework/Versions/3.10/lib/python3.10/_strptime.py", line 568, in _strptime_datetime
tt, fraction, gmtoff_fraction = _strptime(data_string, format)
File "/Library/ManagedFrameworks/Python/Python3.framework/Versions/3.10/lib/python3.10/_strptime.py", line 349, in _strptime
raise ValueError("time data %r does not match format %r" %
ValueError: time data '2023-11-07T05:10:47.' does not match format '%Y-%m-%dT%H:%M:%S.%f'
Nov 09 2023 18:18:56 +0000 WARNING: mdm_status.py return code: 1```

The state.json file looks like this:

```{
  "last_mdm_kickstart": "0001-01-01T00:00:00Z",
  "last_software_update_kickstart": "0001-01-01T00:00:00Z"
}```

**Comment by tuxudo** [comment]
I am not able to duplicate this. What version of the module are you using?

**Comment by ofirgalcon** [comment]
v3.2 - the problem is that in Sonoma MDM Watcher produces this state.json on all the computers I tried so far where at least one of the values is "0001-01-01T00:00:00Z" which seems to confuse the python script.

```{
  {"last_mdm_kickstart":"2023-11-09T17:04:11.590972Z",
"last_software_update_kickstart":"0001-01-01T00:00:00Z"}
}```

**Comment by tuxudo** [comment]
Lines 240 and 242 explicitly check for an invalid timestamp. Using the JSON files you provided in the issue comments, I am not able to duplicate this. Please verify that the affected clients have `"0001-01-01" not in watchdog_state[item]` in their mdm_status.py script

**Comment by tuxudo** [comment]
This should be resolved with v3.3 of the module.
